### PR TITLE
validations and test

### PR DESCRIPTION
**Pull request by Becky449**
- Add the following validations:
- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    it { should have_many(:posts).with_foreign_key('author_id') }
+    it { should have_many(:comments).with_foreign_key('author_id') }
+    it { should have_many(:likes).with_foreign_key('author_id') }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
+    it { should validate_numericality_of(:posts_counter).only_integer.is_greater_than_or_equal_to(0) }
+  end
+end
```
Comment (TadesseDev) [inline spec/models/user_spec.rb]: - [ ] Good job adding the tests @Becky449, But it's among the primary requirements to test every custom method in the models. Hence please add a test for the `three_latest_posts` method which is inside your `user` class.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @Becky449 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
- [ ] Notice the link for your MIT file is broken, Please make sure to add the MIT license file and update its content in regard to your name and project. Also, replace the `[YOUR NAME]` part with your name, and `[APP TYPE]` with your app type.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Becky449 👋

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Highlights 🎯 
✔️ Descriptive PR
✔️ Added validations and model specs 📌 
✔️ No linter errors



 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Just like the previous reviewer suggested, please make sure to add the MIT license file and update its content in regard to your name and project. Also, replace the [YOUR NAME] part with your name, and [APP TYPE] with your app type.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @Becky449 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Blog App -validations and model specs.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**